pkg/dataplane/client.go

```diff
@@ -35,6 +63,63 @@ func NewClient(aud, cloud string, cred azcore.TokenCredential) (*ManagedIdentity
 	return &ManagedIdentityClient{swaggerClient: swaggerClient}, nil
 }
 
+func (c *ManagedIdentityClient) GetUserAssignedMSI(ctx context.Context, request UserAssignedMSIRequest) (*CredentialsObject, error) {
+	validate := validator.New(validator.WithRequiredStructEnabled())
+	if err := validate.Struct(request); err != nil {
+		return nil, fmt.Errorf("%w: %w", errInvalidRequest, err)
+	}
+
+	ctx = context.WithValue(ctx, identityURLKey, request.IdentityURL)
+	credRequestDef := swagger.CredRequestDefinition{
+		IdentityIDs: []*string{&request.ResourceID},
+	}
+
+	creds, err := c.swaggerClient.Getcreds(ctx, credRequestDef, nil)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %w", errGetCreds, err)
+	}
+
+	//
+	// GetCreds can return multiple identities. We expect one. Return if we don't find one.
```

Comment:
under what circumstances will it return multiple?

Reply:
In the `credRequestDef`, you can specify multiple resource IDs to get multiple UA MSI objects. 

Since this method should only get one UA MSI, the "return multiple" scenario would only occur if:
(1) if there is a bug in our swagger client or
(2) MSI-RP has a bug and returns multiple given one input.

We can check if `len(creds.ExplicitIdentities)` is < 1, but I prefer the current approach as a defensive programming practice. It helps identify bugs in our swagger client, and if MSI-RP starts returning multiple UA MSI objects, our RP has access to credentials that it didn't request.

Reply:
in our case we may have multiple UA MSIs, right? Will we need multiple calls or can it be done in one call?

Reply:
We're federating all the UA MSIs except for one of them. The one we don't federate is the one where we need to get the backing certificate. All the others are not assigned to the cluster.

Reply:
To answer your question -> we just have one that MSI per cluster that needs to use this method

Reply:
right. but this is a general library, should we impose ARO's design to the library?

Reply:
anyway, my comment is not blocking. i will approve it.

Reply:
I'll merge now, and we'll add another function for getting multiple UA MSIs when we get that requirement.